waltid-services/waltid-wallet-api/src/main/kotlin/id/walt/webwallet/service/SSIKit2WalletService.kt

```diff
@@ -267,6 +267,36 @@ class SSIKit2WalletService(
         }
         logger.debug { "HTTP Response: $resp, body: $httpResponseBody" }
 
+        var redirectFromBody: String? = null
+        var errorUriFromBody: String? = null
+        if (httpResponseBody != null && httpResponseBody.couldBeJsonObject()) {
+            val bodyJson = httpResponseBody.parseAsJsonObject().getOrNull()
+            val bodyRedirect = bodyJson?.get("redirect_uri")?.jsonPrimitive?.contentOrNull
+            val bodyError = bodyJson?.get("error_uri")?.jsonPrimitive?.contentOrNull
+            @Suppress("HttpUrlsUsage")
+            if (bodyRedirect != null && (bodyRedirect.startsWith("http://") || bodyRedirect.startsWith("https://"))) {
```

Comment:
perhaps startsWIth.. could be extracted into a separate method and used in both cases, body-redirect and body-error
e.g.
```kotlin
fun String.isUrl() = this.startsWith("http://") || this.startsWith("https://")
```
or maybe using regex:
```kotlin
fun String.isUrl(): Boolean {
    val urlPattern = "^https?://.*".toRegex(RegexOption.IGNORE_CASE)
    return urlPattern.matches(this)
}
```
but perhaps better:
```kotlin
fun String.isUrl() = runCatching {
        val url = URI.create(this).toURL()
        url.protocol in listOf("http", "https") && url.host.isNotEmpty()
    }.getOrDefault(false)
```